### DISCUSSION TITLE
Few question before I start making

### DISCUSSION THREAD

**Discussion by shashankx86** [tauri-apps/tauri#8482] [Q&A] [answered]
Hi,

I making a concept app drawer (Don't ask why, i am just doing it)

The idea is to spawn a window(a) without frame outside of desktop windows leaving bit of in side of desktop window
when mouse is placed on window(a) (fig 1) it should slide to specified (like center left right)
 
### fig 1
<img src="https://github.com/slint-ui/slint/assets/64682801/51982f9e-fb3d-49ab-962f-6cca8f63652b" width="50%"> 

### fig 2
<img src="https://github.com/slint-ui/slint/assets/64682801/5c32a242-1fed-43ea-9e88-fff6d63ea3b1" width="50%">

before i start making i want to ask few questions

1. frame less window possible?
2. positioning of window(a) at spawn?
3. sliding of windows with moving animation (i.e i shouldn't just spawn it moving to go at specified area)?

**Comment by FabianLars** [comment]
> I making a concept app drawer (Don't ask why, i am just doing it)

Why? :D

> frame less window possible?

Yes, disable "decorations".

> positioning of window(a) at spawn?

Yes, look for configs and functions with "position" in their name and maybe "center" if that's what you want anyway.

> sliding of windows with moving animation (i.e i shouldn't just spawn it moving to go at specified area)?

I don't understand that but there are no "native" window animation apis in tauri if that answer the question?

**Reply by shashankx86** [reply]
@FabianLars 
One more question 
is it possible for window to detect whether it has mouse over it or not

> Why? :D

idk, maybe i just want to make something


> I don't understand that but there are no "native" window animation apis in tauri if that answer the question?

what i am trying to say is 

[to know: see (fig 1) when mouse pointer goes over it the windows should come at specified location like center, left, etc (see fig 2)]
the windows in fig 1 comes in like fig 2 with like slide animation, it shouldn't just suddenly appear

**Reply by FabianLars** [reply, marked as answer]
>  is it possible for window to detect whether it has mouse over it or not

stuff like the native `mouseover` event should work. Tauri doesn't expose a general input event system yet so using what the webview/browser offers is probably the only way for now. Maybe https://crates.io/crates/rdev too if you can/want to use rust.

> [to know: see (fig 1) when mouse pointer goes over it the windows should come at specified location like center, left, etc (see fig 2)]
the windows in fig 1 comes in like fig 2 with like slide animation, it shouldn't just suddenly appear

Yeah, you can't directly animate windows. You can try to set the window positions in a loop yourself but that may be choppy or weird UX too.
